lib/checkbufferoverrun.cpp

```diff
@@ -2105,7 +2105,8 @@ bool CheckBufferOverrun::analyseWholeProgram(const std::list<Check::FileInfo*> &
                                                    "arrayIndexOutOfBounds",
                                                    CWE788, false);
             errorLogger.reportErr(errmsg);
-            errors = true;
+            if (errorLogger.hasErrors())
```

Comment:
I would like that we put this logic in lib/cppcheck.cpp instead. It's clumpsy that I have to remember this every time I add whole program analysis.


    bool CppCheck::analyseWholeProgram()
    {
        .....
        return errors && errorLogger.hasErrors();
    }

Reply:
I agree with your concern.
But I don't have a clear vision how to put the logic in cppcheck,cpp.

Reply:
hmm.. doesnt it help to modify the existing `CppCheck::analyseWholeProgram()` as I suggested?

Reply:
Aha, got it. Done, please review.